CalPatRec/src/HelixFinder_module.cc

```diff
@@ -0,0 +1,2559 @@
+    // float transVecX = _chColl->at(hitIndice).uDir().y();
+    // float transVecY = -_chColl->at(hitIndice).uDir().x();
+    // float projTransErr = transErr*(transVecX*radVecX + transVecY*radVecY);
+    // _tcHits[tcHitsIndex].circleError2 = projTransErr*projTransErr+projWireErr*projWireErr;
+  }
+}
+
+//-----------------------------------------------------------------------------
+// compute residual between point an circle
+//-----------------------------------------------------------------------------
+float HelixFinder::computeCircleResidual2(size_t& tcHitsIndex, float& xC, float& yC, float& rC) {
+
+  float xP = getPos(tcHitsIndex)->x();
+  float yP = getPos(tcHitsIndex)->y();
+  float deltaDistance = std::abs(rC - std::sqrt((xP - xC) * (xP - xC) + (yP - yC) * (yP - yC)));
+  float circleSigma2 = _tcHits[tcHitsIndex].circleError2;
+
+  return deltaDistance * deltaDistance / circleSigma2;
+}
+
+//-----------------------------------------------------------------------------
+// compute phi relative to helix center, and set helixPhiError
+//-----------------------------------------------------------------------------
+void HelixFinder::computeHelixPhi(size_t& tcHitsIndex, float& xC, float& yC) {
+
+  int hitIndice = _tcHits[tcHitsIndex].hitIndice;
+
+  float X = getPos(tcHitsIndex)->x() - xC;
+  float Y = getPos(tcHitsIndex)->y() - yC;
+  _tcHits[tcHitsIndex].helixPhi = polyAtan2(Y, X);
+  if (_tcHits[tcHitsIndex].helixPhi < 0) {
+    _tcHits[tcHitsIndex].helixPhi = _tcHits[tcHitsIndex].helixPhi + 2 * 3.14;
+  }
+  // find phi error and initialize it
+  // find phi error by projecting errors onto vector tangent to circle
+  float deltaS2(0);
+  if (hitIndice == -1) {
+    deltaS2 = _caloClusterSigma * _caloClusterSigma;
+  } else {
+    float tanVecX = Y / std::sqrt(X * X + Y * Y);
+    float tanVecY = -X / std::sqrt(X * X + Y * Y);
+    float wireErr = _chColl->at(hitIndice).wireRes();
+    float wireVecX = _chColl->at(hitIndice).uDir().x();
+    float wireVecY = _chColl->at(hitIndice).uDir().y();
+    float projWireErr = wireErr * (wireVecX * tanVecX + wireVecY * tanVecY);
+    float transErr = _chColl->at(hitIndice).transRes();
+    float transVecX = _chColl->at(hitIndice).uDir().y();
+    float transVecY = -_chColl->at(hitIndice).uDir().x();
+    float projTransErr = transErr * (transVecX * tanVecX + transVecY * tanVecY);
+    deltaS2 = projWireErr * projWireErr + projTransErr * projTransErr;
+  }
+  _tcHits[tcHitsIndex].helixPhiError2 = deltaS2 / (X * X + Y * Y);
+}
+
+//-----------------------------------------------------------------------------
+// function to compute helix momentum^2 given some circle radius and line slope
+//-----------------------------------------------------------------------------
+float HelixFinder::computeHelixMomentum2(float& radius, float& dphidz) {
+
+  float lambda = 1.0 / dphidz;
+
+  return (radius * radius + lambda * lambda) * (3.0 / 10) * (3.0 / 10) * _bz0 * _bz0;
```

Comment:
Please use CLHEP::c_light/1000.0 to convert from mm to MeV/c.  I define this as:
KinKal::BFieldMap::cbar(); if you want to use that.